### DISCUSSION TITLE
array `.includes` method fails in TypeScript when the array members are zod enum members

### DISCUSSION THREAD

**Discussion by samislam** [colinhacks/zod#3163] [Q&A] [answered]
In TypeScript, the following code is valid:
```ts
const c = "BMW"
const isOK = ["apple", "banana", "strawberry"].includes("BMW")
```
However, when using Zod, it's not valid:
```ts
const AccountTypeEnum = z.enum(['regular', 'hotspotMAC', 'mikrotikACL', 'starOSACL', 'DOCSIS'])

const c = "BMW"
const isOK = [
  AccountTypeEnum.enum.regular,
  AccountTypeEnum.enum.hotspotMAC,
  AccountTypeEnum.enum.DOCSIS)
].includes(c)
```
It gives the following error:
```ts
Argument of type '"BMW"' is not assignable to parameter of type '"regular" | "hotspotMAC" | "DOCSIS"'.ts(2345)
```

Am I doing something wrong? how to overcome this issue?

**Comment by JacobWeisenburger** [comment, marked as answer]
`stringArray` is of type `string[]`, so it is theoretically possible to include any string there. 

```ts
const stringArray = [ 'apple', 'banana', 'strawberry' ]
// const stringArray: string[]
```

`unionOfStringsArray` is of type `('regular' | 'hotspotMAC' | 'DOCSIS')[]`, so it is not possible to include `BMW`.

```ts
const AccountTypeEnum = z.enum( [ 'regular', 'hotspotMAC', 'mikrotikACL', 'starOSACL', 'DOCSIS' ] )

const unionOfStringsArray = [
    AccountTypeEnum.enum.regular,
    AccountTypeEnum.enum.hotspotMAC,
    AccountTypeEnum.enum.DOCSIS,
]
// const unionOfStringsArray: ('regular' | 'hotspotMAC' | 'DOCSIS')[]
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by samislam** [comment]
The array of enum values includes string literals and is not `string[]`, thus, it's showing an error, therefore I will have to convert the literal string types to simply `string[]`.

The solution is to do casting with `as string[]`, or if you want:
```ts
const unionOfStringsArray = [
    AccountTypeEnum.enum.regular,
    AccountTypeEnum.enum.hotspotMAC,
    AccountTypeEnum.enum.DOCSIS,
].map(String).includes("BMW")
```
